### ISSUE TITLE
Resolution Doesn't Revert Because It Can't Detect Windows Process Closing

### ISSUE THREAD

**Issue by Blootrix** [cgarst/gamestream_launchpad#24]
Launchpad will not revert my resolution back to the previous resolution when exiting because it can't detect Playnite closing. Launchpad stays open in the tray after closing Playnite and has to be manually closed, resolution and refresh rate then need to be manually reset.
As far as I can tell, the reason is that it's not reading the Windows process correctly.

launcher_window_name = Playnite
Nothing reverts because it can't detect Playnite closing.

launcher_window_name = Playnite.FullscreenApp.exe
Launchpad closes Playnite before it can fully open and reverts resolution back, I'm assuming because of the way Playnite opens its fullscreen mode.

Hopefully this helps to figure out why resolutions aren't reverting back after closing and you can tell us what to target instead for those that aren't having any joy with "launcher_window_name = Playnite"

Thanks!

**Comment by Blootrix** [comment]
Did some more testing and using the two lines below, if I manually close Playnite then everything works.
I've not been able to find a way to stop the stream via Moonlight and revert it by ending the stream which closes the app, but this solution is the best I'll be able to figure out.

launcher_window_name = Playnite.FullscreenApp.exe
& 
close_watch_method = process

**Comment by ReenigneCA** [comment]
if you use the close_method playnite_mutex it will properly detect when playnite closes though you'll lose fixing resolution changes that happen if a game doesn't revert the resolution back. If you don't have complicated needs hdr_moonlight_launcher now supports resolution and refresh rate changes and gracefully shuts down gamestream. The issue with shutting down gamestream is that it's based on detecting a window being created and then closed which gamesteam launchpad doesn't do. If you use another launcher in addition (like hdr_moonlight_launcher which starts gamestream launchpad,) the first program that is run is the one whose window is important to shut down the session. Anyway I recommend using hdr_moonlight_launcher and having it handle resolution and refresh rate changes and then if you need additional functionality using gamestream_launchpad.

**Comment by Blootrix** [comment]
> if you use the close_method playnite_mutex it will properly detect when playnite closes though you'll lose fixing resolution changes that happen if a game doesn't revert the resolution back. If you don't have complicated needs hdr_moonlight_launcher now supports resolution and refresh rate changes and gracefully shuts down gamestream. The issue with shutting down gamestream is that it's based on detecting a window being created and then closed which gamesteam launchpad doesn't do. If you use another launcher in addition (like hdr_moonlight_launcher which starts gamestream launchpad,) the first program that is run is the one whose window is important to shut down the session. Anyway I recommend using hdr_moonlight_launcher and having it handle resolution and refresh rate changes and then if you need additional functionality using gamestream_launchpad.

Thanks for the reply and info.

I've tried googling it and found a GitHub for a Moonlight HDR Launcher, is that what you were talking about?

**Comment by ReenigneCA** [comment]
yeah sorry I named it wrong https://github.com/lyckantropen/moonlight_hdr_launcher It was made to enable hdr in games that support HDR but don't specifically support gamestream. Turning on HDR used to be all it did but I didn't like waiting for gamestream launchpad to load up so I submitted a patch that enables changing display mode and refresh rate and added a small window that closes once the program you run closes so it actually shuts down gamestream gracefully. The actual HDR toggle can be left on it will do nothing if you don't have an HDR display or turned off to not bother. If you get an EDID emulator you can use HDR on an HDR capable TV even if your computer monitors don't support it. Anyway it doesn't have some of the more advanced features of gamestream like it doesn't support running a programs before the main launcher and doesn't do any of the py autgui stuff like minimizing other windows but since it's a C++ executable it loads way faster so your screen should be setup by the time you actually start seeing things in Moonlight.
